src/xdp-app-info.c

```diff
@@ -1000,8 +1000,6 @@ xdp_connection_create_app_info_sync (GDBusConnection  *connection,
 
   cache_insert_app_info (sender, app_info);
 
-  xdp_connection_track_name_owners (connection, on_peer_died);
```

Comment:
Turns out that this is a pretty bad thing to do. It subscribes to a signal every time a new client connects and never disconnects it. It fixes https://github.com/flatpak/xdg-desktop-portal/issues/1416.